rust/lance-datafusion/src/udf.rs

```diff
@@ -26,8 +26,116 @@ pub fn register_functions(ctx: &SessionContext) {
     ctx.register_udf(json::json_get_bool_udf());
     ctx.register_udf(json::json_array_contains_udf());
     ctx.register_udf(json::json_array_length_udf());
+
+    // GEO functions
+    ctx.register_udf(ST_INTERSECTS_UDF.clone());
+    ctx.register_udf(ST_WITHIN_UDF.clone());
+    ctx.register_udf(BBOX_UDF.clone());
+}
+
+static ST_WITHIN_UDF: LazyLock<ScalarUDF> = LazyLock::new(st_within);
+static BBOX_UDF: LazyLock<ScalarUDF> = LazyLock::new(bbox);
+static ST_INTERSECTS_UDF: LazyLock<ScalarUDF> = LazyLock::new(st_intersects);
+
+fn st_intersects() -> ScalarUDF {
```

Comment:
I think we should directly use functions in `geodatafusion`

Reply:
if that Pr is merged its just rebase. Also I remember there was some issue in datafusion python api where mixed case functions couldn't be used.